docs/websocket/choice_recognitions.md

```diff
@@ -70,30 +46,36 @@ organisation_id | `string` | **Required** The id of the organisation in which to
 challenge_id    | `string` | **Required** The id of the challenge to prepare.
 
 
-## Stream recognition audio
+## Register audio procedure
 
-The audio to be recognised can now be streamed to the server. The streaming
-works by repeatedly calling this RPC. Each time the RPC is called, the data
-will be appended to an audio file on the server.
+To send audio for recognition a WAMP procedure must be registered. Check your
+WAMP library's documentation on how to do this. Some are linked here:
 
-### URI
+* [Autobahn-js][1]
+* [Autobahn|Python][2]
 
-```
-nl.itslanguage.choice.write
-```
+The procedure must return [progressive results], check your library's
+documentation on how to implement this. Every progressive result is a chunk of
+the audio and so audio is sent in chunks.
+Later on, the server will call this procedure to 'pull' the audio in. When the
+procedure returns, the stream is finished/closed.
 
-### Parameters
+Note that a complete (self-describing) audio file is expected, starting with a
+header. Check [the audio documentation page](audio.md#self-describing-formats)
+to see more about supported formats.
 
-Name           | Type     | Description
----------------|----------|------------
-recognition_id | `string` | **Required** The id of the recognition for which to upload the audio.
-data           | `bytes`  | **Required** The data to stream to the server. This can be any length.
-encoding       | `string` | **Optional** The encoding in which the data is sent. This should be `base64` if the data is base64 encoded.
+The procedure can best be registered to a randomly generated URI, using uuid4
+for this is advisable to ensure uniqueness. For every audio a separate procedure
+must be registered. The URI is used later on.
+It is advisable to unregister the procedure when the streaming is finished.
+
+!!! warning
+    Re-using a URI can result in conflicts.
 
 
 ## Perform the recognition
 
-After completing the streaming of the audio, the recognition can be performed.
+The recognition can be performed when all preparations are done.
```

Comment:
perhaps it makes sense to add a new heading/paragraph on the 'dance' we're expecting client/server to perform before diving into to details without explaining the overall idea. It might be easier to grok the concept:
* call initialise audio 
* register rpc, stream audio
* start to recognise audio
(this might no longer reflect current reality but you get the idea, right ;-)

Reply:
Added it to the top of the file ;)